File: typical90/024.rs
use proconio::input;

fn main() {
    input! {
        n: usize,
        k: usize,
        a: [i32; n],
        b: [i32; n],
    }

    let is_possible = |diff: i32, k: i32| -> bool {
        diff <= k && diff % 2 == k % 2
    };

    let mut diff = 0;
    for i in 0..n {
        diff += (a[i]-b[i]).abs();
    }

    if is_possible(diff, k as i32) {
        println!("Yes");
    } else {
        println!("No");
    }
}

File: typical90/C++/024.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,n) for (int i = 0; i < (n); ++i)
using ll = long long;

bool solve() {
  int n, k;
  cin >> n >> k;
  vector<int> a(n), b(n);
  rep(i,n) cin >> a[i];
  rep(i,n) cin >> b[i];

  ll diff = 0;
  rep(i,n) diff += abs(a[i]-b[i]);

  if (diff > k) return false;
  if (diff%2 != k%2) return false;
  return true;
}

int main() {
  cout << (solve() ? "Yes" : "No");
  return 0;
}